www/components/messagebox.tsx

```diff
@@ -1,28 +1,74 @@
+import { useCallback, useState } from "react";
 import Image from "next/image";
 import icon from "@/public/bloomicon.jpg";
 import usericon from "@/public/usericon.svg";
 import Skeleton from "react-loading-skeleton";
+import { FaLightbulb, FaThumbsDown, FaThumbsUp } from "react-icons/fa";
+import { API } from "@/utils/api";
+import useSWR, { mutate } from "swr";
 
-interface MessageBoxRegularProps {
-  children: React.ReactNode;
+interface MessageBoxProps {
   isUser?: boolean;
-  loading?: false;
+  messageId?: string;
+  conversationId?: string;
+  api?: API;
+  text: string;
+  loading?: boolean;
+  isThoughtsOpen?: boolean;
+  setIsThoughtsOpen: (isOpen: boolean) => void;
+  setThought: (thought: string) => void;
 }
 
-interface MessageBoxLoadingProps {
-  children?: React.ReactNode;
-  isUser?: boolean;
-  loading: true;
-}
-
-// merge the two types
-type MessageBoxProps = MessageBoxRegularProps | MessageBoxLoadingProps;
+const fetchThought = async (key: string) => {
+  const [_, conversationId, messageId, userId] = key.split(":");
+  const api = new API({ url: process.env.NEXT_PUBLIC_API_URL!, userId });
+  return api.getThoughtById(conversationId, messageId);
+};
 
 export default function MessageBox({
-  children,
   isUser,
-  loading,
+  messageId,
+  api,
+  text,
+  loading = false,
+  setIsThoughtsOpen,
+  conversationId,
+  setThought,
 }: MessageBoxProps) {
+  const [shouldFetch, setShouldFetch] = useState(false);
+  const shouldShowButtons = messageId !== "";
+
+  const {
+    data: thought,
+    error,
+    isValidating,
+  } = useSWR(
+    shouldFetch && messageId && api
+      ? `thought:${conversationId}:${messageId}:${api.userId}`
+      : null,
+    fetchThought,
+    {
+      revalidateOnFocus: false,
+      revalidateIfStale: false,
+      revalidateOnReconnect: false,
+      onSuccess: (data) => {
+        console.log(data);
+        if (data) {
+          setIsThoughtsOpen(true);
+          setThought(data);
+        }
+      },
+    },
+  );
+
+  const handleFetchThought = useCallback(() => {
+    setShouldFetch(true);
+    if (messageId && api) {
+      const key = `thought:${conversationId}:${messageId}:${api.userId}`;
+      mutate(key);
+    }
+  }, [messageId, api, conversationId]);
```

Comment:
The useCallback seems a bit unnecessary here. The method isn't being passed to any child components and I don't see why any of those dependencies would change in the lifecycle of a `MessageBox` component.

Reply:
Good call!